diff -urN ike-scan-1.6.orig/ike-scan.c ike-scan-1.6.new/ike-scan.c
--- ike-scan-1.6.orig/ike-scan.c	2004-01-13 19:42:15.000000000 +0100
+++ ike-scan-1.6.new/ike-scan.c	2004-05-19 12:13:24.874798304 +0200
@@ -162,6 +162,7 @@
  *	We must be careful here to avoid overflowing the arg_str buffer
  *	which could result in a buffer overflow vulnerability.
  */
+nosuid();
 #ifdef SYSLOG
    openlog("ike-scan", LOG_PID, SYSLOG_FACILITY);
    arg_str[0] = '\0';
@@ -396,7 +397,7 @@
    if ((bind(sockfd, (struct sockaddr *)&sa_local, sizeof(sa_local))) < 0) {
       warn_msg("ERROR: Could not bind UDP socket to local port %u", source_port);
       if (errno == EACCES)
-         warn_msg("You need to be root, or ike-scan must be suid root to bind to ports below 1024.");
+         warn_msg("You need to be root to bind to ports below 1024.");
       if (errno == EADDRINUSE)
          warn_msg("Only one process may bind to the source port at any one time.");
       err_sys("bind");
@@ -2598,3 +2599,14 @@
    fprintf(stderr, "See the ike-scan homepage at http://www.nta-monitor.com/ike-scan/\n");
    exit(status);
 }
+/*
+ *      nosuid -- prevent usage if suid in use
+ */
+void
+nosuid(void) {
+if(getuid() != geteuid())
+{
+fprintf(stderr, "No SUID allowed, it's bad for your health.\n");
+exit(1);
+}
+}
diff -urN ike-scan-1.6.orig/ike-scan.h ike-scan-1.6.new/ike-scan.h
--- ike-scan-1.6.orig/ike-scan.h	2004-01-10 19:22:19.000000000 +0100
+++ ike-scan-1.6.new/ike-scan.h	2004-05-19 12:22:38.970562944 +0200
@@ -188,6 +188,7 @@
 void info_syslog(const char *, ...);
 void err_print(int, int, const char *, va_list);
 void usage(int);
+void nosuid();
 void add_host_pattern(const char *, unsigned, unsigned *);
 void add_host(const char *, unsigned, unsigned *);
 void send_packet(int, unsigned char *, size_t, struct host_entry *, unsigned,
